code/src/utils/calculate-cart-items-counter/CalculateCartItemsCounter.test.ts

```diff
@@ -0,0 +1,39 @@
+import { CartItem } from "@/types/cart.types";
+import calculateCartItemsCounter from "@/utils/calculate-cart-items-counter/CalculateCartItemsCounter";
+
+const generateCart = (length: number, increaseQuantity: boolean = false) => {
+  return Array.from({ length }, (_, i) => ({
+    productId: i + 1,
+    quantity: increaseQuantity ? i + 1 : 1
+  })) as unknown as CartItem[];
+};
+
+describe("Test CalculateCartItemsCounter", () => {
+  test("Should return 0 when cart is empty", () => {
+    const cart = [] as unknown as CartItem[];
+
+    const result = calculateCartItemsCounter(cart);
+    expect(result).toBe(0);
+  });
+
+  test("Should return 6 when cart has 3 items with summ quantity equal 6", () => {
+    const cart = generateCart(3, true);
+
+    const result = calculateCartItemsCounter(cart);
+    expect(result).toBe(6);
+  });
+
+  test("Should return 99 when cart has 99 items with quantity 1 each", () => {
+    const cart = generateCart(99);
+
+    const result = calculateCartItemsCounter(cart);
+    expect(result).toBe(99);
+  });
+
+  test("Should return 99+ when cart has 150 items", () => {
+    const cart = generateCart(150);
+
+    const result = calculateCartItemsCounter(cart);
+    expect(result).toBe("99+");
+  });
```

Comment:
I suggest to use `test.each` here because tests are rather similar

https://dev.to/bgord/simplify-repetitive-jest-test-cases-with-test-each-310m

Reply:
Thanks, used `test.each` there